app/assets/javascripts/pageflow/editor/api/file_types.js

```diff
@@ -29,26 +29,42 @@ pageflow.FileTypes = pageflow.Object.extend({
   },
 
   setup: function(serverSideConfigs) {
+    var clientSideConfigs = this.clientSideConfigs;
     this._setup = true;
 
-    this.fileTypes = _.map(serverSideConfigs, function(serverSideConfig) {
-      var clientSideConfig = this.clientSideConfigs[serverSideConfig.collectionName];
-
-      if (!clientSideConfig) {
-        throw 'Missing client side config for file type "' + serverSideConfig.collectionName + '"';
+    this.fileTypes = new pageflow.
+      FileTypesCollection(_.map(serverSideConfigs, function(serverSideConfig) {
+        var clientSideConfig = clientSideConfigs[serverSideConfig.collectionName];
+
+        if (!clientSideConfig) {
+          throw 'Missing client side config for file type "' +
+            serverSideConfig.collectionName + '"';
+        }
+
+        _(this.clientSideConfigModifications[serverSideConfig.collectionName])
+          .each(function(modification) {
+            this.lintModification(modification, serverSideConfig.collectionName);
+            this.applyModification(clientSideConfig, modification);
+          }, this);
+
+        return new pageflow.FileType(_.extend({}, serverSideConfig, clientSideConfig));
+      }, this));
+
+    _.map(this.fileTypes.fileTypes, function(fileType) {
+      if(fileType && fileType.nestedFileTypes.fileTypes){
+        fileType.nestedFileTypes = new pageflow.FileTypesCollection(
+          _.map(fileType.nestedFileTypes.fileTypes, function(nestedFileType) {
```

Comment:
The names of the nested file types are already included in the config seed data generated by the server, because in FileTypes#each, we recursively search for file types of nested files, starting at the top level file types.

This is what the editor receives:
```
"fileTypes":[{"collectionName":"image_files","typeName":"Pageflow::ImageFile","paramKey":"image_file","i18nKey":"pageflow/image_file"},{"collectionName":"video_files","typeName":"Pageflow::VideoFile","paramKey":"video_file","i18nKey":"pageflow/video_file"},{"collectionName":"audio_files","typeName":"Pageflow::AudioFile","paramKey":"audio_file","i18nKey":"pageflow/audio_file"},{"collectionName":"pageflow_panorama_packages","typeName":"Pageflow::Panorama::Package","paramKey":"package","i18nKey":"pageflow/panorama/package"},{"collectionName":"text_track_files","typeName":"Pageflow::TextTrackFile","paramKey":"text_track_file","i18nKey":"pageflow/text_track_file"}],
```

However, this data is not structured according to the nesting hierarchy. Maybe we can make this easier to read by also passing the editor this kind of nesting information, or by extracting the nested file types in a way that is more similar to top level file type extraction.